website/content/en/guides/level-up/csv-enrichment-guide.md

```diff
@@ -0,0 +1,210 @@
+IoT device that emits its status - `online`, `offline`, `transmitting`, and
+`error` states, but the device needs to emit as little payload as possible
+due to technical constraints or requirements; instead, the IoT can device can
+emit integers — `1` for `online`, `2` for `offline`, etc.
+
+To enrich the IoT device's observability data, let's use a `csv` file
+containing the following information:
+
+```csv
+status_code,status_message
+1,"device status online"
+2,"device status offline"
+3,"device status connection error"
+...
+7,"device status transmitting"
+8,"device status transmission complete"
+9,"device status not responding"
+```
+
+To enrich your observability data, you can use ['get_enrichment_table_record'][get_enrichment_table_record].
+Assuming that your `csv` file is called `iot_status.csv`, the following
+illustrates the required Vector configuration:
+
+``` toml
+[enrichment_tables.iot_status]
+type = "file"
+
+[enrichment_tables.iot_status.file]
+path = "/etc/vector/iot_status.csv"
+encoding = { type = "csv" }
+
+[enrichment_tables.iot_status.schema]
+status_code = "integer"
+status_message = "string"
+```
+
+After this configuration, we can now translate the output from IoT devices to
+human-readable messages that provide further context in our `iot_status.csv`.
+To do so, we can make use of the [`get_enrichment_table_record`][get_enrichment_table_record] function:
+
+``` toml
+[transforms.enrich_iot_status]
+type = "remap"
+inputs = ["datadog_agents"]
+source = '''
+. = parse_json!(.status_message)
+
+status_code = del(.status_code)
+
+row = get_enrichment_table_record!("iot_status", {"status_code" : status_code})
+
+.status = row.status_message
+'''
+```
+
+Your observability data, assuming it was in JSON, now has been transformed from:
+
+```json
+{
+  "host":"my.host.com",
+  "timestamp":"2019-11-01T21:15:47+00:00",
+  ...
+  "status_code":1,
+}
+```
+
+To:
+
+```json
+{
+  "host":"my.host.com",
+  "timestamp":"2019-11-01T21:15:47+00:00",
+  ...
+  "status":"device status transmission complete",
+}
+```
+
+While the previous example is relatively straightforward, the second example
+use case really shows how powerful event enrichment can be for your
+observability pipeline.
+
+## Use Case 2 - Setting Alerts for Suspicious Access
+
+In this example, you are dealing with a system where attempted access from
+specific identifiers, such as a blacklisted IP address, must automatically trigger
+alerts to relevant personnel on your team. You can use an external database
+(though Vector's current solution is limited to `csv` files) that contain
+a list of blacklisted IP address and enrich the data so that whatever downstream
+log management solution your team may be using, such as
+[Datadog's Log Management product], can trigger the alert based on the scrubbed
+log.
+
+The key benefit to using `enrichment tables` in this case is that you can
+enrich your observability data to trigger an alert on your on-premise network
```

Comment:
I don't think this is specific to on-premises infrastructure, even though we've seen that use case where files with sensitive data need to be kept locally. I would add it more as a note than as part of the core use case.